internal/db/code.go

```diff
@@ -0,0 +1,114 @@
+package db
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/starlight-go/starlight"
+	"reflect"
+)
+
+type Code struct {
+	ID       uuid.UUID
+	Name     string
+	Function Function
+	Runtime  Runtime
+	Code     string
+}
+
+type Runtime int64
+
+const (
+	Golang Runtime = iota
+	Javascript
+	Starlark
+)
+
+type Function int64
+
+const (
+	Validator Function = iota
+)
+
+var nativeFunctionMap map[Code]func(interface{}) (interface{}, error) = map[Code]func(interface{}) (interface{}, error){
+	boolValidator:         boolValidatorFunc,
+	intValidator:          intValidatorFunc,
+	enumValidator:         enumValidatorFunc,
+	stringValidator:       stringValidatorFunc,
+	textValidator:         textValidatorFunc,
+	emailAddressValidator: emailAddressValidatorFunc,
+	uuidValidator:         uuidValidatorFunc,
+	floatValidator:        floatValidatorFunc,
+	URLValidator:          URLValidatorFunc,
+}
+
+func CallFunc(c Code, sf StorageFormat, arg interface{}) (interface{}, error) {
+	if c.Runtime == Golang {
+		if f, ok := nativeFunctionMap[c]; ok {
+			return f(arg)
+		} else {
+			return nil, fmt.Errorf("Func %s not found in native functions", c.Name)
+		}
+	} else if c.Runtime == Starlark {
+		return skylarkParser(c.Code, sf, arg)
+	}
+	return nil, nil
+}
+
+//Starlark
+//uses https://github.com/starlight-go/starlight
+func skylarkParser(code string, sf StorageFormat, arg interface{}) (interface{}, error) {
```

Comment:
I'm not going to review this implementation for now -- I assume this is easy enough to rewrite to be a bit more generic for other `Function`s